tests/models/multimodal/test_config_validation.py

```diff
@@ -0,0 +1,439 @@
+    batch[key_to_check] = np.zeros(tuple(wrong_time_shape))
+
+    match_str = (
+        rf"'{key_to_check}' shape error using interval .* Expected B x \{{.*\}}, Got \{{.*\}}"
+    )
+    with pytest.raises(ValueError, match=match_str):
+        validate(batch, config)
+
+
+def test_validate_batch_error_wrong_shape_spatial(
+    valid_config_dict, sample_numpy_batch
+):
+    """Test ValueError for incorrect spatial dimension size."""
+    config = deepcopy(valid_config_dict)
+    batch = deepcopy(sample_numpy_batch)
+    key_to_check = "satellite_actual"
+
+    if not config.get("sat_encoder"):
+        pytest.skip("sat_encoder not in config fixture.")
+    if key_to_check not in batch and config.get("sat_encoder"):
+        pytest.skip(
+            f"Config expects {key_to_check}, but sample_numpy_batch missing it."
+        )
+    elif key_to_check not in batch:
+        pytest.skip(f"{key_to_check} not in batch, cannot test wrong spatial shape.")
+
+    correct_shape = batch[key_to_check].shape
+    if len(correct_shape) < 3:
+        pytest.skip("Batch has insufficient dimensions.")
+    wrong_spatial_shape = list(correct_shape)
+    wrong_spatial_shape[2] += 1
+    batch[key_to_check] = np.zeros(tuple(wrong_spatial_shape))
+
+    match_str = (
+        rf"'{key_to_check}' shape error using interval .* Expected B x \{{.*\}}, Got \{{.*\}}"
+    )
+    with pytest.raises(ValueError, match=match_str):
+        validate(batch, config)
+
+
+def _get_batch_size_test_helper(batch_dict, key):
+    """Helper (for testing) to get batch size, handling NWP dict structure."""
+    if key not in batch_dict:
+        raise KeyError(f"Key '{key}' not found in batch for batch size check.")
+    data = batch_dict[key]
+
+    if key == "nwp":
+        if not isinstance(data, dict) or not data:
+            raise TypeError(f"NWP data for key '{key}' in batch is not a non-empty dict")
+        first_source_key = next(iter(data.keys()))
+        first_source_dict = data[first_source_key]
+        if not isinstance(first_source_dict, dict):
+            raise TypeError(
+                f"NWP source value for '{first_source_key}' is not a dict: {first_source_dict}"
+            )
+        try:
+            data_array = next(
+                v
+                for v in first_source_dict.values()
+                if isinstance(v, np.ndarray) and v.ndim > 0
+            )
+        except StopIteration:
+            raise ValueError(
+                f"No numpy array with ndim > 0 found within NWP source dict: {first_source_dict}"
+            )
+        return data_array.shape[0]
+    elif isinstance(data, np.ndarray):
+        if data.ndim == 0:
+            raise ValueError(f"Data for key '{key}' is a 0-dim array, cannot get batch size.")
+        return data.shape[0]
+    else:
+        raise TypeError(
+            f"Data for key '{key}' ({type(data)}) is not a numpy array or NWP dict."
+        )
+
+
+def test_validate_batch_error_batch_size_inconsistency(
```

Comment:
change this to `test_validate_batch_size`, 
- in validate function put in batch size
- check batch size when checking shapre
- this test can check errors are raised